sympy/solvers/solveset.py

```diff
@@ -3051,7 +3053,11 @@ def nonlinsolve(system, *symbols):
                 return result
 
         # positive dimensional system
-        return _handle_positive_dimensional(polys, symbols, denominators)
+        res = _handle_positive_dimensional(polys, symbols, denominators)
+        if isinstance(res, EmptySet) and [not p.domain.is_Exact for p in polys]:
```

Comment:
Why a list instead of `any()`? They seem to give different results.
```
>>> any(False for i in range(2))
False
>>> bool([False for i in range(2)])
True
```

Reply:
I am sorry. In a hurry I missed that one. I'll just make it correct.